spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/JedisConnectionConfiguration.java

```diff
@@ -95,6 +96,19 @@ private JedisClientConfigurationBuilder applyProperties(JedisClientConfiguration
 		return builder;
 	}
 
+	private void createPoolingBuilder(JedisClientConfigurationBuilder builder, Environment environment, Pool pool) {
+		final boolean poolEnabled = environment.getProperty("spring.redis.jedis.pool.enabled", Boolean.class, true);
```

Comment:
Given that `spring.redis.jedis.pool.enabled` is a property that you read in code, there's no reason to not map it in `Pool`. This will remove all those `Environment` injection points above.